src/components/LetterCasePicker.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC, memo, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { titleCase } from 'title-case'
+import { formatLetterCaseActionCreator as formatLetterCase } from '../actions/formatLetterCase'
+import { isTouch } from '../browser'
+import useWindowOverflow from '../hooks/useWindowOverflow'
+import getThoughtById from '../selectors/getThoughtById'
+import themeColors from '../selectors/themeColors'
+import fastClick from '../util/fastClick'
+import head from '../util/head'
+import TriangleDown from './TriangleDown'
+import LowerCaseIcon from './icons/LowerCaseIcon'
+import SentenceCaseIcon from './icons/SentenceCaseIcon'
+import TitleCaseIcon from './icons/TitleCaseIcon'
+import UpperCaseIcon from './icons/UpperCaseIcon'
+
+/** Letter Case Picker component. */
+const LetterCasePicker: FC<{ fontSize: number; style?: React.CSSProperties }> = memo(({ fontSize, style }) => {
+  const colors = useSelector(themeColors)
+  const ref = useRef<HTMLDivElement>(null)
+  const dispatch = useDispatch()
+  const overflow = useWindowOverflow(ref)
+
+  /** Toggles the Letter Case to the clicked swatch. */
+  const toggleLetterCase = (command: string, e: React.MouseEvent | React.TouchEvent) => {
+    e.stopPropagation()
+    e.preventDefault()
+    dispatch(formatLetterCase(command))
+  }
+  const selected = useSelector(state => {

```

Comment:
This selector uses a lot of the same logic as `getUpdatedThoughtValue` in the action-creator. This could lead to discrepancies in the future. Instead, move `getUpdatedThoughtValue` into a separate util function that both modules can use.